cmd/ttn-lw-cli/commands/gateways.go

```diff
@@ -434,41 +435,76 @@ var (
 	gatewaysConnectionStats = &cobra.Command{
 		Use:     "get-connection-stats [gateway-id]",
 		Aliases: []string{"connection-stats", "cnx-stats", "stats"},
-		Short:   "Get connection stats for a gateway",
+		Short:   "Get connection stats for a (group of) gateway(s).",
+		Long: `Get connection stats for a (group of) gateway(s).
+		Use the --gateway-ids flag for multiple gateways.
+		If both the parameter and the flag are provided, the parameter is ignored.`,
 		RunE: func(cmd *cobra.Command, args []string) error {
-			gtwID, err := getGatewayID(cmd.Flags(), args, true)
+			ids, err := cmd.Flags().GetStringSlice("gateway-ids")
 			if err != nil {
 				return err
 			}
 
-			is, err := api.Dial(ctx, config.IdentityServerGRPCAddress)
+			gtwID, err := getGatewayID(cmd.Flags(), args, true)
 			if err != nil {
-				return err
+				if !errors.IsInvalidArgument(err) || len(ids) == 0 {
+					return err
+				}
 			}
 
-			gateway, err := ttnpb.NewGatewayRegistryClient(is).Get(ctx, &ttnpb.GetGatewayRequest{
-				GatewayIds: gtwID,
-				FieldMask:  ttnpb.FieldMask("gateway_server_address"),
-			})
+			is, err := api.Dial(ctx, config.IdentityServerGRPCAddress)
 			if err != nil {
 				return err
 			}
 
-			if gsMismatch := compareServerAddressGateway(gateway, config); gsMismatch {
-				return errAddressMismatchGateway.New()
-			}
-
 			gs, err := api.Dial(ctx, config.GatewayServerGRPCAddress)
 			if err != nil {
 				return err
 			}
 
-			res, err := ttnpb.NewGsClient(gs).GetGatewayConnectionStats(ctx, gtwID)
-			if err != nil {
-				return err
+			var result any
+			if len(ids) > 0 {
+				paths, err := cmd.Flags().GetStringSlice("paths")
+				if err != nil {
+					return err
+				}
+
+				var gtwIDs []*ttnpb.GatewayIdentifiers
+				for _, id := range ids {
+					gtwIDs = append(gtwIDs, &ttnpb.GatewayIdentifiers{
+						GatewayId: id,
+					})
+				}
+				res, err := ttnpb.NewGsClient(gs).BatchGetGatewayConnectionStats(ctx,
+					&ttnpb.BatchGetGatewayConnectionStatsRequest{
+						GatewayIds: gtwIDs,
+						FieldMask: &types.FieldMask{
+							Paths: paths,
+						},
+					})
+				if err != nil {
+					return err
+				}
+				result = res
```

Comment:
Why not just `return io.Write(os.Stdout, config.OutputFormat, res)`? Then you also don't need the `else`.